azurelinuxagent/ga/exthandlers.py

```diff
@@ -297,8 +292,10 @@ def _incarnation_changed(self, etag):
         return self.last_etag != etag
 
     def __last_gs_unsupported(self):
+
         # Return if the last GoalState was unsupported
-        return self.__gs_aggregate_status.status == GoalStateStatus.Failed and \
+        return self.__gs_aggregate_status is not None and \
```

Comment:
consider raising when it is None

Reply:
discussed offline, leaving it as is for now